### PR TITLE
Upgrade to react-native@v0.31.0

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

File: android/app/src/main/java/org/mozilla/magnet/notificationlistener/NotificationListener.java [hunk 1]
```diff
@@ -36,6 +40,18 @@ public NotificationListener(ReactApplicationContext reactContext, Activity activ
                     new IntentFilter("notification-delete"));
     }
 
+    @ReactMethod
+    public void appLaunchedFromNotification(Promise promise) {
```
Comment (arcturus) [inline android/app/src/main/java/org/mozilla/magnet/notificationlistener/NotificationListener.java:44]: I though we have a metric to check this right?

Comment (arcturus) [reply android/app/src/main/java/org/mozilla/magnet/notificationlistener/NotificationListener.java:44]: Forget about it, realised that is is native land :S

File: android/build.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.3'
+        classpath 'com.android.tools.build:gradle:2.2.1'
```
Comment (arcturus) [inline android/build.gradle]: I have the feeling that this change is making the travis build fail.
Any reason for the update?

**Comment by wilsonpage** [comment]
@arcturus r?

**Review by arcturus — APPROVED** [review]
LGTM, now that the gradlew is working.

**Pull request merged**